### ISSUE TITLE
[Modern] bug when updating json scalar field

### ISSUE THREAD

**Issue by jardakotesovec** [facebook/relay#2049]
I am using  [JSON Scalar field](https://github.com/taion/graphql-type-json).

And getting error:
```
recycleNodesInto.js:48 Uncaught (in promise) TypeError: Cannot assign to read only property 'finishingGroupIds' of object '#<Object>'
    at recycleNodesInto.js:48
```

when relay store receives update from the mutation with new json scalar value, which has internally additional item in the array. When I reload the page - store interpret the json value just fine, so its just issue when replacing the old json value with new one.

This whole recycleNodesInto function already seems to be doing some magic for flow, so not sure where is actually the root of the problem. Maybe just some condition to avoid stepping into the json scalar field?

I could work on PR if you give me some hints how you think this should be addressed.

![screenshot 2017-08-21 10 57 52](https://user-images.githubusercontent.com/1590681/29511555-c7ec68f0-865f-11e7-9ace-f6a110e3b072.png)

**Comment by alexjgraham** [comment]
I've noticed this since updating to `1.2.0`

**Comment by dbslone** [comment]
I've been running into the same issue with JSON fields when upgrading our app to Relay Modern. First occurred in `1.1.0` and upgraded to `1.2.0` and the issue still exists.

After lots of debugging I found that the reason for the failure appears to be the `deepFreeze` in development mode. If you set a breakpoint on line 48 and in the console type: `Object.isFrozen(nextObject['data'])` it will return true

**Comment by dbslone** [comment]
@jardakotesovec PR #2051 solves the issue for me. Not sure if it will be approved but for now it sovles the issue.

**Comment by mike-marcacci** [comment]
For anybody else running into this, I've temporarily published to @boltline/relay-runtime. Until this or another fix get merged I'll keep this in sync with Facebook's relay-runtime releases. Once a fix is permanently in place, I'll deprecate all releases of my fork on NPM.

**Comment by jstejada** [comment]
@jardakotesovec, does this only happen in DEV mode?

**Comment by bejito** [comment]
Hi ! I am facing the same issue. Any chance this is being fixed soon ? Or is there a way to fix this in dev mode ?

**Comment by josephsavona** [comment]
Yup, fixed in v2.0.0

**Comment by jstejada** [comment]
Closing; if the issue persists please let us know and we’ll reopen the issue
